docs: add everything is an event tutorial (#2495)
aquasecurity/tracee: docs/tutorials/everything-is-an-event.md

From:

```
- event=anti_debugging,aslr_inspection,cgroup_notify_on_release,cgroup_release_agent,core_pattern_modification,default_loader_mod,disk_mount,docker_abuse,dynamic_code_loading,fileless_execution,hidden_file_created,illegitimate_shell,k8s_api_connection,k8s_cert_theft,kernel_module_loading,ld_peload,process_vm_write_inject,proc_fops_hooking,proc_kcore_read,proc_mem_access,proc_mem_code_injection,ptrace_code_injection,rcd_modification,sched_debug_recon,scheduled_task_mod,stdio_over_socket,sudoers_modification,syscall_hooking,system_request_key_mod
```

To (we added `execve` to the end of the list):
 
```
- event=anti_debugging,aslr_inspection,cgroup_notify_on_release,cgroup_release_agent,core_pattern_modification,default_loader_mod,disk_mount,docker_abuse,dynamic_code_loading,fileless_execution,hidden_file_created,illegitimate_shell,k8s_api_connection,k8s_cert_theft,kernel_module_loading,ld_peload,process_vm_write_inject,proc_fops_hooking,proc_kcore_read,proc_mem_access,proc_mem_code_injection,ptrace_code_injection,rcd_modification,sched_debug_recon,scheduled_task_mod,stdio_over_socket,sudoers_modification,syscall_hooking,system_request_key_mod,execve
``` 

> Let's restart Tracee, so it starts with the new configuration.

Reviewer: ```suggestion
Let's restart Tracee to run it with the new configuration.
```

# Everything is an event

This guide will show an example of running the new tracee experience we are experimenting with on Kubernetes.

> For more context on this change, please check the dicussion on github: LINK

Reviewer 2: 1. typo: discussion
2. LINK should be replaced with a proper link

Author: Yes, the discussion was not created yet, it is a placeholder

## Prerequisites  
  
minikube - see installation instructions and dependencies [here](https://minikube.sigs.k8s.io/docs/start/). 
 

> *Note: Your local cluster must be run on an intel processor. Apple silicon is not currently supported.* 

Reviewer 2: we support ARM64 as well

Author: Meant for minikube, not tracee

kube-system   kube-apiserver-minikube            1/1     Running   0          26s  
kube-system   kube-controller-manager-minikube   1/1     Running   0          26s  
kube-system   kube-proxy-cvqjm                   1/1     Running   0          15s  
kube-system   kube-scheduler-minikube            1/1     Running   0          26s  
kube-system   storage-provisioner                1/1     Running   0          15s  
``` 
 
## Running tracee as a daemon set 

Tracee is installed as a daemonset, and the new experience we are trying is not enabled by default.
To install we can use helm and pass a custom flag `everythingIsAnEvent`.
 
``` 
helm install tracee aqua/tracee  --namespace tracee-system --create-namespace  -set everythingIsAnEvent=true 

``` 
 
You can verify that the tracee daemon set is running via: 
 
```  
kubectl get pods   
NAME           READY   STATUS    RESTARTS   AGE  
tracee-fcjmp   1/1     Running   0          4m11s  
``` 
  
## Tracee in action  
 
At this point tracee is running with new behavior. You can see this at the top of the pod’s logs:
 
``` 
kubectl logs -f daemonset/tracee -n tracee-system 
``` 

> To test it we can trigger a rule, like the anti_debugging one. 

Reviewer 2: We keep giving the anti-debugging example, but this example is not a good one. Maybe give another (e.g. fileless)?

Author: Yes, but I didn’t have time to prepare the necessary docker image dor a different attack. There is a task for it. Do you have a simple filessimage or instructions we can do here?

Author: Is it okay if we use Carolina example? https://github.com/krol3/demo-fileless#demo-using-tracee-and-postee

Reviewer 2: Yes, looks like a good example.

``` 

Let's restart tracee, so it starts with the new configuration.

``` 
kubectl rollout restart ds/tracee -n tracee-system 

``` 

Now if you check the logs you will see the `execve` events that are happening on your cluster.

``` 
kubectl  logs -f ds/tracee -n tracee-system | grep execve 
 

{"timestamp":1671035008366697931,"threadStartTime":1776459241935,"processorId":5,"processId":26734,"cgroupId":20433,"threadId":26734,"parentProcessId":21685,"hostProcessId":26734,"hostThreadId":26734,"hostParentProcessId":21685,"userId":1000,"mountNamespace":4026531841,"pidNamespace":4026531836,"processName":"bash","hostName":"josedonizetti-x","containerId":"","containerImage":"","containerName":"","podName":"","podNamespace":"","podUID":"","eventId":"59","eventName":"execve","argsNum":2,"returnValue":0,"stackAddresses":null,"contextFlags":{"containerStarted":false},"args":[{"name":"pathname","type":"const char*","value":"/usr/bin/grep"},{"name":"argv","type":"const char*const*","value":["grep","--color=auto","execve"]}]} 

``` 

Tracee supports lots of events, you can see a list by running:

```
kubectl  exec ds/tracee -n tracee-system -- /tracee/tracee --list
```

> Try to experiment with them and let us know what you think on the github discussion or slack: LINK

Reviewer 2: LINK should be updated with a proper link

Author: Yup, it is a placeholder, discussion was not created yet
